### PR TITLE
Attribute, Rating and Stock filters: Fix the dropdown indicator icon display

### LINKED ISSUE #7889: Hide the arrow in dropdown if there's no more options to choose from
## Describe the bug

When there's no more options to choose from the filter dropdown, the arrow is still visible there. We should hide the input, otherwise it feels like you can still add more options, but clicking on the control doesn't do anything.

## To reproduce

1. Add the Filter by Rating to the page (make sure you have at least one product with rating)
2. In Editor set it to "Dropdown" and "Multiple"
3. Go to frontend
4. Choose all of the available option

## Expected behavior

There's no indicator there's still something to choose from

## Actual behavior

The arrow is still there, but clicking on the control doesn't do anything.

## Screenshots

https://user-images.githubusercontent.com/3616980/206177760-73419348-0f61-4fd6-a52a-5700a5cc140a.webm

## Additional context

1. The issue may be already fixed in the newer versions of Gutenberg, but at the moment woocoomerce-blocks uses outdated version (current wordpress-components version: 14.2.0). Please check if the version has been bumped and if the bug still occurs at the moment of picking up this issue.

2. Issue can be reproduced with the Dropdown set to "single". It occurs differently: no arrow, but when clicked, there's an input available, without any option to choose.

### LINKED ISSUE #8075: Filter blocks with dropdown display style should also show the chevron when Single selection is enabled
## To reproduce

1. Create a post or page with the Filter by Stock, Filter by Rating and Products blocks.
2. Select the Filter by Stock Controls block and choose: _Single_ and _Dropdown_ in the Display Settings.
3. Select the Filter by Rating Controls block and choose: _Multiple_ and _Dropdown_ in the Display Settings.
4. Go to the frontend.
5. Notice the Filter by Stock dropdown doesn't have a chevron on the right, while the Filter by Rating block does have it.

## Expected behavior

Both dropdowns should display the chevron to signal to the user that clicking on them will open a list of suggestions.

## Additional context

The relevant code can be found here:

https://github.com/woocommerce/woocommerce-blocks/blob/32d4605c4582a97e7a4c806884b98034044de3d7/assets/js/blocks/attribute-filter/block.tsx#L669-L671

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR fixes the display logic of the dropdown indicator icon (chevron) for the `Single` option of `Allow selecting multiple options?` setting. Fixes #8075

Additionally, it also hides the chevron there are no more options to choose from. Fixes #7889
### Screenshots

| Before | After |
| ------ | ----- |
|Products_–_ratings-2|Products_–_ratings|

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Add the Attribute, Rating and Stock filter blocks.
2. Set the `Display Style` to `Dropdown` and `Allow selecting multiple options?` to `Single`.
3. Ensure the dropdown chevron displays correctly both in the editor and the page.
4. Ensure there's no conflict between the cancel 'x' icon and the chevron when dropdown options are selected.
5. With the `Allow selecting multiple options?` set to `Multiple` test selecting all of the options and check if the chevron is correctly hidden.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Attribute, Rating and Stock filters: Dropdown indicator icon display fix.

### CODE REVIEW

File: assets/js/blocks/attribute-filter/block.tsx [hunk 1]
```diff
@@ -666,7 +666,7 @@ const AttributeFilterBlock = ( {
 								),
 							} }
 						/>
-						{ multiple && (
+						{ checked.length === 0 && ! isLoading && (
```
Comment (kmanijak) [inline assets/js/blocks/attribute-filter/block.tsx]: That change covers the case where dropdown is set to Single choice, but it little bug for dropdowns with Multiple choice: when user selected at least one option, the chevron would be hidden, even though there's more options to choose from. So this case should also be covered. Maybe something among those lines:

```suggestion
						{ multiple || checked.length === 0 && ! isLoading && (
```

There's also [this issue](https://github.com/woocommerce/woocommerce-blocks/issues/7889) about hiding chevron if there's no more options to choose from in Multiple choice dropdowns, which now I think it's much easier than I thought at the time I created the issue 🙈 I think the above suggestion could be extended with:

```
const showChevron = multiple ?
  checked.length < displayedOptions.length : // MULTIPLE: show chevron only when not every option is chosen
  checked.length === 0                       // SINGLE: show chevron only when none option is chosen
```

to cover each case.

But if you feel that's out of scope of this PR it's fine and you can just stick to the first suggestion 🙌

**Review by kmanijak — CHANGES_REQUESTED** [review]
Thanks for working on this. As I wrote in a comment I spotted one issue with the Multiple version of the Dropdowns.

If you feel like you want, you can also cover [this issue](https://github.com/woocommerce/woocommerce-blocks/issues/7889) 💪

**Review by kmanijak — APPROVED** [review]
Works like a charm 🎉 

Considering that you fixed https://github.com/woocommerce/woocommerce-blocks/issues/7889 as well, you can mention it in the description (so the issue will be closed) and extend the testing steps, so it's covered during release.

But anyway, approving! 👍

**Review by danielwrobert — APPROVED** [review]
I also tested and everything works according to the test plan!

It would be nice to be able to click the input field when the dropdown is open a second time to close it, as opposed to clicking outside the field (see below). That is beyond the scope of this PR and I can create an Issue to investigate separately – I mainly wanted to make a note of it since I noticed while testing. 🙂

CleanShot 2023-01-04 at 12 17 41

Nice work! 🚀

**Pull request merged**